src/theme/useThemeToggle.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { darkMode, lightMode, currentTheme } from '../common/constants';
+
+export const useThemeToggle = () => {
+  const [theme, setTheme] = useState(lightMode);
+
+  const setMode = mode => {
+    window.localStorage.setItem(currentTheme, mode);
+    setTheme(mode);
+  };
+
+  const toggleTheme = () => {
+    if (theme === lightMode) {
+      setMode(darkMode);
```

Comment:
theme === lightMode... ? do see sometihng wrong here... Again you are binding your hook to the fact that you will have only **two**! **modes**!. In reality you need to be able to operate with **multiple**! **themes**!

Think about how this code should be refactroed in order to be flexible enough to support multiple themes in future. But still be able to be used for your toggle